packages/core/src/modules/proofs/ProofsApi.ts

```diff
@@ -270,6 +317,46 @@ export class ProofsApi<
     return proofRecord
   }
 
+  /**
+   * Answer with a new presentation request in response to received presentation proposal message
+   * to the connection associated with the proof record.
+   *
+   * @param options multiple properties like proof record id, proof formats to accept requested credentials object
+   * specifying which credentials to use for the proof
+   * @returns Proof record associated with the sent request message
+   */
+  public async requestProofAsResponse(options: RequestProofAsResponseOptions<PFs>): Promise<ProofExchangeRecord> {
+    const { proofRecordId } = options
+
+    const proofRecord = await this.getById(proofRecordId)
+
+    const service = this.getService(proofRecord.protocolVersion)
+
+    if (!proofRecord.connectionId) {
+      throw new AriesFrameworkError(
+        `No connectionId found for credential record '${proofRecord.id}'. Connection-less issuance does not support negotiation.`
+      )
+    }
+
+    const connection = await this.connectionService.getById(this.agentContext, proofRecord.connectionId)
+
+    // Assert
+    connection.assertReady()
+
+    const requestOptions: CreateRequestAsResponseOptions<PFs> = {
+      proofRecord,
+      proofFormats: options.proofFormats,
+      autoAcceptProof: options.autoAcceptProof,
+      comment: options.comment,
+    }
+    const { message } = await service.createRequestAsResponse(this.agentContext, requestOptions)
+
+    const outbound = createOutboundMessage(connection, message)
```

Comment:
createOutboundMessage has been replaced by the class OutboundMessageContext

Reply:
Fixed